Bottom: 1ff3d419bbd678b27790929be1720a45a797225f
Top:    15bc4ccc76f3f588234e9fdf4a35288722134f6d
Author: scott snyder <sss@karma>
Date:   2020-01-06 20:55:27 -0500

Refresh of AGirlandHerFed

---

diff --git a/ComicReader/src/com/blogspot/applications4android/comicreader/comics/AGirlandHerFed.java b/ComicReader/src/com/blogspot/applications4android/comicreader/comics/AGirlandHerFed.java
index a333025..0ec88cf 100755
--- a/ComicReader/src/com/blogspot/applications4android/comicreader/comics/AGirlandHerFed.java
+++ b/ComicReader/src/com/blogspot/applications4android/comicreader/comics/AGirlandHerFed.java
@@ -2,12 +2,12 @@ package com.blogspot.applications4android.comicreader.comics;
 
 import java.io.BufferedReader;
 import java.io.IOException;
-import com.blogspot.applications4android.comicreader.comictypes.IndexedComic;
+import com.blogspot.applications4android.comicreader.comictypes.RandomIndexedComic;
 import com.blogspot.applications4android.comicreader.core.Strip;
 import com.blogspot.applications4android.comicreader.exceptions.ComicLatestException;
 
 
-public class AGirlandHerFed extends IndexedComic {
+public class AGirlandHerFed extends RandomIndexedComic {
 
 	@Override
 	protected String getFrontPageUrl() {
@@ -53,6 +53,22 @@ public class AGirlandHerFed extends IndexedComic {
 	}
 
 	@Override
+	protected int parseForPrevId(String line, int def) {
+		if(line == null) {
+			return def;
+		}
+                return getIdFromStripUrl (line);
+	}
+
+	@Override
+	protected int parseForNextId(String line, int def) {
+		if(line == null) {
+			return def;
+		}
+                return getIdFromStripUrl (line);
+	}
+
+        @Override
 	protected boolean htmlNeeded() {
 		return true;
 	}
@@ -78,4 +94,59 @@ public class AGirlandHerFed extends IndexedComic {
 		strip.setText("-NA-");
 		return "https://agirlandherfed.com/"+final_str;
 	}
+
+
+        @Override
+	protected int getNextStripId(BufferedReader br, String url) {
+          String str;
+          try {
+            while ((str = br.readLine()) != null) {
+              int i = str.indexOf("next.png");
+              if (i != -1) {
+                int i2 = str.lastIndexOf ("href=\"1.", i);
+                if (i2 != -1) {
+                  i2 += 8;
+                  int i3 = str.indexOf (".html", i2);
+                  if (i3 != -1) {
+                    return Integer.parseInt (str.substring (i2, i3));
+                  }
+                }
+              }
+            }
+          }
+          catch (Exception e) {
+            e.printStackTrace();
+          }
+          return 0;
+        }
+
+        @Override
+	protected int getPreviousStripId(BufferedReader br, String url) {
+          String str;
+          try {
+            while ((str = br.readLine()) != null) {
+              int i = str.indexOf("back.png");
+              if (i != -1) {
+                int i2 = str.lastIndexOf ("href=\"1.", i);
+                if (i2 != -1) {
+                  i2 += 8;
+                  int i3 = str.indexOf (".html", i2);
+                  if (i3 != -1) {
+                    return Integer.parseInt (str.substring (i2, i3));
+                  }
+                }
+              }
+            }
+          }
+          catch (Exception e) {
+            e.printStackTrace();
+          }
+          return 0;
+        }
+
+	@Override
+	protected String getRandUrl() {
+          // Not implemented.
+          return getStripUrlFromId (1);
+	}
 }
